File: kid-ios/MyWorld/Parent/StatsView.swift
import SwiftUI
import Charts
import UIKit

/// PRD §4.5 stats hub — four focused sub-pages, each visible the moment its
/// data exists. The old single-scroll view buried everything; this routes
/// to the right surface for the question the parent is asking.
struct StatsView: View {
    @Environment(AuthManager.self) private var auth

    var body: some View {
        ScrollView {
            VStack(spacing: 12) {
                hubCard(icon: "chart.xyaxis.line",      title: "Usage over time",
                        subtitle: "Taps per category, day by day",
                        destination: AnyView(UsageView()))
                hubCard(icon: "chart.bar.fill",         title: "Top words",
                        subtitle: "Most-tapped words this month",
                        destination: AnyView(TopWordsView()))
                hubCard(icon: "magnifyingglass",        title: "Word history",
                        subtitle: "Search every tap, by word and time",
                        destination: AnyView(WordHistoryView()))
                hubCard(icon: "gauge.with.dots.needle.67percent",
                        title: "Game accuracy",
                        subtitle: "Pass rate by category and by game mode",
                        destination: AnyView(AccuracyView()))
                hubCard(icon: "text.bubble.fill",
                        title: "Sentence activity",
                        subtitle: "What they said with the sentence builder",
                        destination: AnyView(SentenceActivityView()))
                hubCard(icon: "hand.raised.fingers.spread.fill",
                        title: "How they answer",
                        subtitle: "Tap · verbal · object · physical · gesture",
                        destination: AnyView(InputMethodsView()))
                hubCard(icon: "rosette",                title: "Mastery & sessions",
                        subtitle: "30-day mastery and recent activity",
                        destination: AnyView(MasterySessionsView()))
            }
            .padding(16)
        }
        .background(Color(hex: Brand.bg))
        .navigationTitle("Stats")
        .navigationBarTitleDisplayMode(.inline)
    }

    private func hubCard(icon: String, title: String, subtitle: String,
                         destination: AnyView) -> some View {
        NavigationLink { destination } label: {
            HStack(spacing: 14) {
                Image(systemName: icon)
                    .font(.system(size: 22, weight: .semibold))
                    .foregroundStyle(.white)
                    .frame(width: 50, height: 50)
                    .background(Color(hex: Brand.pink), in: Circle())
                VStack(alignment: .leading, spacing: 3) {
                    Text(title)
                        .font(.system(size: 17, weight: .bold, design: .rounded))
                        .foregroundStyle(Color(hex: Brand.ink))
                    Text(subtitle)
                        .font(.system(size: 12))
                        .foregroundStyle(Color(hex: Brand.muted))
                        .multilineTextAlignment(.leading)
                }
                Spacer()
                Image(systemName: "chevron.right")
                    .font(.system(size: 14, weight: .semibold))
                    .foregroundStyle(Color(hex: Brand.faint))
            }
            .padding(16)
            .frame(maxWidth: .infinity)
            .background(Color(hex: Brand.card), in: RoundedRectangle(cornerRadius: 16))
            .overlay(RoundedRectangle(cornerRadius: 16).stroke(Color(hex: Brand.line), lineWidth: 1))
        }
        .buttonStyle(.plain)
    }
}


// MARK: -- Sentence activity (what the sentence builder actually said)

/// Weekly sentence count + delta, per-day bars, and the recent sentences with
/// timestamps — the moments a parent actually wants to read. (Lives in this
/// file so no xcodegen re-run.)
struct SentenceActivityView: View {
    @Environment(AuthManager.self) private var auth
    private let api = APIClient()

    @State private var data: APIClient.AnalyticsResponse?
    @State private var errorText: String?

    var body: some View {
        ScrollView {
            VStack(spacing: 14) {
                if let d = data {
                    if let s = d.sentences {
                        summaryCard(s)
                        recentCard(s)
                    } else {
                        emptyState
                    }
                } else if let e = errorText {
                    Text(e).font(.footnote).foregroundStyle(.red)
                } else {
                    ProgressView("Loading sentences…").padding(.top, 60)
                }
            }
            .padding(16)
            // Full width from the FIRST frame — without this the loading
            // spinner defines the content width and the page renders as a
            // skinny centered column that pops wide when the cards land.
            .frame(maxWidth: .infinity, minHeight: UIScreen.main.bounds.height * 0.8, alignment: .top)
        }
        .background(Color(hex: Brand.bg))
        .navigationTitle("Sentence activity")
        .navigationBarTitleDisplayMode(.inline)
        .task {
            do { data = try await api.analytics(childId: auth.childSlug) }
            catch { errorText = "Could not load: \(error.localizedDescription)" }
        }
    }

    private var emptyState: some View {
        VStack(spacing: 10) {
            Image(systemName: "text.bubble")
                .font(.system(size: 34))
                .foregroundStyle(Color(hex: Brand.faint))
            Text("No sentences yet")
                .font(.system(size: 17, weight: .bold, design: .rounded))
                .foregroundStyle(Color(hex: Brand.ink))
            Text("When the sentence builder is on and your child plays a sentence with ▶, it lands here: what they said, and when.")
                .font(.footnote)
                .foregroundStyle(Color(hex: Brand.muted))
                .multilineTextAlignment(.center)
        }
        .padding(.top, 60)
        .padding(.horizontal, 24)
    }

    private struct DayBar: Identifiable { let id: Int; let name: String; let n: Int }

    private func summaryCard(_ s: APIClient.AnalyticsResponse.SentencesPayload) -> some View {
        let diff = s.weekCount - s.prevWeekCount
        let fmt = DateFormatter()
        fmt.dateFormat = "EEE"
        let bars: [DayBar] = s.perDay.enumerated().map { i, n in
            let daysAgo = s.perDay.count - 1 - i
            let d = Calendar.current.date(byAdding: .day, value: -daysAgo, to: Date()) ?? Date()
            return DayBar(id: i, name: daysAgo == 0 ? "today" : fmt.string(from: d), n: n)
        }
        return VStack(alignment: .leading, spacing: 12) {
            Text("Sentences this week")
                .font(.system(size: 18, weight: .bold, design: .rounded))
                .foregroundStyle(Color(hex: Brand.pinkDeep))
            HStack(alignment: .firstTextBaseline, spacing: 10) {
                Text("\(s.weekCount)")
                    .font(.system(size: 40, weight: .heavy, design: .rounded))
                    .foregroundStyle(Color(hex: Brand.ink))
                if diff != 0 {
                    Text(diff > 0 ? "↑ \(diff) vs last week" : "↓ \(-diff) vs last week")
                        .font(.system(size: 13, weight: .bold))
                        .foregroundStyle(diff > 0 ? Color(hex: Brand.good) : .orange)
                } else if s.prevWeekCount > 0 {
                    Text("same as last week")
                        .font(.system(size: 13, weight: .semibold))
                        .foregroundStyle(Color(hex: Brand.muted))
                }
            }
            Chart(bars) { b in
                BarMark(x: .value("Day", b.name), y: .value("Sentences", b.n))
                    .foregroundStyle(Color(hex: Brand.pink))
                    .cornerRadius(4)
            }
            .chartYAxis {
                AxisMarks(position: .leading) { _ in
                    AxisGridLine()
                    AxisValueLabel().font(.system(size: 9))
                }
            }
            .frame(height: 120)
        }
        .padding(14)
        .frame(maxWidth: .infinity, alignment: .leading)
        .background(Color(hex: Brand.card), in: RoundedRectangle(cornerRadius: 16))
        .overlay(RoundedRectangle(cornerRadius: 16).stroke(Color(hex: Brand.line), lineWidth: 1))
    }

    private func recentCard(_ s: APIClient.AnalyticsResponse.SentencesPayload) -> some View {
        VStack(alignment: .leading, spacing: 12) {
            Text("Recent sentences")
                .font(.system(size: 18, weight: .bold, design: .rounded))
                .foregroundStyle(Color(hex: Brand.pinkDeep))
            if s.recent.isEmpty {
                Text("Nothing in the last two weeks.")
                    .font(.footnote)
                    .foregroundStyle(Color(hex: Brand.muted))
            } else {
                ForEach(s.recent) { r in
                    HStack(alignment: .firstTextBaseline) {
                        Text("“\(r.text)”")
                            .font(.system(size: 15, weight: .semibold, design: .rounded))
                            .foregroundStyle(Color(hex: Brand.ink))
                        Spacer()
                        Text(prettyWhen(r.at))
                            .font(.system(size: 12))
                            .foregroundStyle(Color(hex: Brand.faint))
                    }
                    .padding(.vertical, 8)
                    .padding(.horizontal, 10)
                    .background(Color(hex: "#fff7fb"), in: RoundedRectangle(cornerRadius: 10))
                }
            }
        }
        .padding(14)
        .frame(maxWidth: .infinity, alignment: .leading)
        .background(Color(hex: Brand.card), in: RoundedRectangle(cornerRadius: 16))
        .overlay(RoundedRectangle(cornerRadius: 16).stroke(Color(hex: Brand.line), lineWidth: 1))
    }

    private func prettyWhen(_ iso: String?) -> String {
        guard let iso else { return "" }
        let p = ISO8601DateFormatter()
        p.formatOptions = [.withInternetDateTime, .withFractionalSeconds]
        let d = p.date(from: iso) ?? ISO8601DateFormatter().date(from: iso)
        guard let d else { return "" }
        let f = DateFormatter()
        f.dateFormat = "MMM d, h:mm a"
        return f.string(from: d)
    }
}

// MARK: -- Usage over time (parity with the web dashboard's "Use" chart)

/// Taps per category over time — the web dashboard's headline "Use" chart,
/// now native. Day (14) / week (8) / month (6) buckets, top categories as
/// lines, names prettified. (Lives in this file so no xcodegen re-run.)
struct UsageView: View {
    @Environment(AuthManager.self) private var auth
    private let api = APIClient()

    @State private var data: APIClient.AnalyticsResponse?
    @State private var bucket = "day"
    @State private var errorText: String?

    var body: some View {
        ScrollView {
            VStack(spacing: 14) {
                Picker("Granularity", selection: $bucket) {
                    Text("Day").tag("day")
                    Text("Week").tag("week")
                    Text("Month").tag("month")
                }
                .pickerStyle(.segmented)

                if let d = data {
                    usageCard(d.use.series, labels: d.labels)
                } else if let e = errorText {
                    Text(e).font(.footnote).foregroundStyle(.red)
                } else {
                    ProgressView("Loading usage…").padding(.top, 60)
                }
            }
            .padding(16)
            .frame(maxWidth: .infinity, minHeight: UIScreen.main.bounds.height * 0.8, alignment: .top)
        }
        .background(Color(hex: Brand.bg))
        .navigationTitle("Usage over time")
        .navigationBarTitleDisplayMode(.inline)
        .task(id: bucket) { await load() }
    }

    private func load() async {
        do { data = try await api.analytics(childId: auth.childSlug, bucket: bucket) }
        catch { errorText = "Could not load: \(error.localizedDescription)" }
    }

    private struct Pt: Identifiable {
        var id: String { category + "-" + String(i) }
        let category: String
        let i: Int
        let taps: Int
    }

    private func usageCard(_ series: [APIClient.AnalyticsResponse.UseSeries],
                           labels: [String]) -> some View {
        // Top 8 categories by total taps — same cap as the web chart.
        let top = series.sorted { $0.data.reduce(0, +) > $1.data.reduce(0, +) }.prefix(8)
        let points: [Pt] = top.flatMap { s in
            s.data.enumerated().compactMap { i, n in
                guard i < labels.count else { return nil }
                return Pt(category: prettySkillName(s.name), i: i, taps: n)
            }
        }
        return VStack(alignment: .leading, spacing: 12) {
            Text("Taps per category")
                .font(.system(size: 18, weight: .bold, design: .rounded))
                .foregroundStyle(Color(hex: Brand.pinkDeep))
            Text("How often \(prettyChildName(auth.user?.slug).isEmpty ? "your child" : prettyChildName(auth.user?.slug)) communicates with each category on their own.")
                .font(.footnote)
                .foregroundStyle(Color(hex: Brand.muted))
            if points.allSatisfy({ $0.taps == 0 }) || points.isEmpty {
                Text("No board activity yet. This lights up as the board gets used.")
                    .font(.footnote)
                    .foregroundStyle(Color(hex: Brand.muted))
                    .padding(.vertical, 10)
            } else {
                Chart(points) { p in
                    LineMark(x: .value("When", p.i), y: .value("Taps", p.taps))
                        .foregroundStyle(by: .value("Category", p.category))
                        .interpolationMethod(.catmullRom)
                }
                .chartLegend(position: .bottom, alignment: .leading, spacing: 8)
                .chartXAxis {
                    AxisMarks(values: stride(from: 0, to: labels.count, by: max(1, labels.count / 6)).map { $0 }) { v in
                        AxisGridLine()
                        AxisValueLabel {
                            if let i = v.as(Int.self), i < labels.count {
                                Text(labels[i]).font(.system(size: 9))
                            }
                        }
                    }
                }
                .chartYAxis {
                    AxisMarks(position: .leading) { _ in
                        AxisGridLine()
                        AxisValueLabel().font(.system(size: 9))
                    }
                }
                .frame(height: 260)
            }
        }
        .padding(14)
        .frame(maxWidth: .infinity, alignment: .leading)
        .background(Color(hex: Brand.card), in: RoundedRectangle(cornerRadius: 16))
        .overlay(RoundedRectangle(cornerRadius: 16).stroke(Color(hex: Brand.line), lineWidth: 1))
    }
}

File: android-native/app/src/main/java/io/andrewpeterson/myworld/ui/parent/StatsView.kt
package io.andrewpeterson.myworld.ui.parent

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material3.CircularProgressIndicator
import androidx.compose.material3.HorizontalDivider
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.LaunchedEffect
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.compose.ui.window.Dialog
import androidx.compose.ui.window.DialogProperties
import io.andrewpeterson.myworld.LocalAppContainer
import io.andrewpeterson.myworld.model.prettyChildName
import io.andrewpeterson.myworld.net.AnalyticsResponse
import io.andrewpeterson.myworld.net.analytics
import io.andrewpeterson.myworld.ui.theme.Brand
import io.andrewpeterson.myworld.ui.theme.hexColor
import io.andrewpeterson.myworld.ui.theme.prettySkillName

/**
 * The stats hub — port of `Parent/StatsView.swift`: six focused sub-pages
 * behind one full-screen dialog with its own back navigation.
 */
@Composable
fun StatsView(onDismiss: () -> Unit) {
    var page by remember { mutableStateOf<String?>(null) }

    Dialog(onDismissRequest = onDismiss, properties = DialogProperties(usePlatformDefaultWidth = false)) {
        Column(Modifier.fillMaxSize().background(hexColor("#fff7fb"))) {
            // Top bar: back within the hub, or close.
            Row(
                Modifier.fillMaxWidth().background(Brand.pink).padding(horizontal = 14.dp, vertical = 12.dp),
                verticalAlignment = Alignment.CenterVertically,
            ) {
                Text(
                    if (page == null) "✕" else "‹ Back",
                    fontSize = 17.sp, fontWeight = FontWeight.Bold, color = Color.White,
                    modifier = Modifier.clickable { if (page == null) onDismiss() else page = null }
                        .padding(6.dp),
                )
                Spacer(Modifier.width(10.dp))
                Text(pageTitle(page), fontSize = 18.sp, fontWeight = FontWeight.Bold, color = Color.White)
            }

            when (page) {
                null -> StatsHub { page = it }
                "usage" -> UsagePage()
                "topwords" -> TopWordsPage()
                "history" -> WordHistoryPage()
                "accuracy" -> AccuracyPage()
                "inputs" -> InputMethodsPage()
                "mastery" -> MasterySessionsPage()
            }
        }
    }
}

private fun pageTitle(page: String?): String = when (page) {
    "usage" -> "Usage over time"
    "topwords" -> "Top words"
    "history" -> "Word history"
    "accuracy" -> "Game accuracy"
    "inputs" -> "How they answer"
    "mastery" -> "Mastery & sessions"
    else -> "Stats"
}

@Composable
private fun StatsHub(onOpen: (String) -> Unit) {
    val rows = listOf(
        Triple("usage", "📈", "Usage over time" to "Taps per category, day by day"),
        Triple("topwords", "🏆", "Top words" to "Most-tapped words this month"),
        Triple("history", "🔎", "Word history" to "Search every tap, by word and time"),
        Triple("accuracy", "🎯", "Game accuracy" to "Pass rate by category and by game mode"),
        Triple("inputs", "🖐", "How they answer" to "Tap · verbal · object · physical · gesture"),
        Triple("mastery", "🏅", "Mastery & sessions" to "30-day mastery and recent activity"),
    )
    Column(
        Modifier.fillMaxSize().verticalScroll(rememberScrollState()).padding(16.dp),
    ) {
        rows.forEach { (id, emoji, texts) ->
            Row(
                Modifier.fillMaxWidth()
                    .background(Color.White, RoundedCornerShape(16.dp))
                    .border(1.dp, hexColor("#f3c6dd"), RoundedCornerShape(16.dp))
                    .clickable { onOpen(id) }
                    .padding(16.dp),
                verticalAlignment = Alignment.CenterVertically,
            ) {
                Box(Modifier.size(50.dp).background(Brand.pink, CircleShape), contentAlignment = Alignment.Center) {
                    Text(emoji, fontSize = 22.sp)
                }
                Spacer(Modifier.width(14.dp))
                Column(Modifier.weight(1f)) {
                    Text(texts.first, fontSize = 17.sp, fontWeight = FontWeight.Bold, color = Brand.ink)
                    Text(texts.second, fontSize = 12.sp, color = Brand.muted)
                }
                Text("›", fontSize = 20.sp, color = Brand.muted)
            }
            Spacer(Modifier.height(12.dp))
        }
    }
}

// ── Shared loader chrome ────────────────────────────────────────────────────

@Composable
fun LoadingSpinner(label: String) {
    Column(
        Modifier.fillMaxWidth().padding(top = 60.dp),
        horizontalAlignment = Alignment.CenterHorizontally,
    ) {
        CircularProgressIndicator(color = Brand.pink)
        Spacer(Modifier.height(10.dp))
        Text(label, fontSize = 13.sp, color = Brand.muted)
    }
}

// ── Usage over time (the web dashboard's headline "Use" chart) ──────────────

@Composable
private fun UsagePage() {
    val c = LocalAppContainer.current
    var bucket by remember { mutableStateOf(0) }   // 0=day 1=week 2=month
    var data by remember { mutableStateOf<AnalyticsResponse?>(null) }
    var error by remember { mutableStateOf<String?>(null) }

    LaunchedEffect(bucket) {
        data = null; error = null
        try {
            data = c.api.analytics(c.auth.childSlug, listOf("day", "week", "month")[bucket])
        } catch (e: Exception) { error = "Could not load: ${e.message}" }
    }

    Column(Modifier.fillMaxSize().verticalScroll(rememberScrollState()).padding(16.dp)) {
        SegmentedChips(listOf("Day" to 0, "Week" to 1, "Month" to 2), bucket) { bucket = it }
        Spacer(Modifier.height(14.dp))
        val d = data
        when {
            error != null -> Text(error!!, fontSize = 13.sp, color = Color(0xFFDC2626))
            d == null -> LoadingSpinner("Loading usage…")
            else -> StatCard("Taps per category") {
                Text(
                    "How often ${prettyChildName(c.auth.childSlug).ifEmpty { "your child" }} communicates with each category on their own.",
                    fontSize = 12.sp, color = Brand.muted,
                )
                Spacer(Modifier.height(10.dp))
                // Top 8 categories by total taps — same cap as the web chart.
                val top = d.useSeries.sortedByDescending { it.data.sum() }.take(8)
                val series = top.map { s ->
                    ChartSeries(prettySkillName(s.name), s.data.map { it.toFloat() })
                }
                if (series.isEmpty() || series.all { s -> s.values.all { it == 0f } }) {
                    Text("No board activity yet. This lights up as the board gets used.",
                        fontSize = 12.sp, color = Brand.muted,
                        modifier = Modifier.padding(vertical = 10.dp))
                } else {
                    MultiLineChart(series, d.labels, chartHeight = 260.dp)
                }
            }
        }
    }
}

// ── Game accuracy: by category + by game mode ───────────────────────────────

@Composable
private fun AccuracyPage() {
    val c = LocalAppContainer.current
    var data by remember { mutableStateOf<AnalyticsResponse?>(null) }
    var error by remember { mutableStateOf<String?>(null) }

    LaunchedEffect(Unit) {
        try { data = c.api.analytics(c.auth.childSlug) }
        catch (e: Exception) { error = "Could not load: ${e.message}" }
    }

    Column(Modifier.fillMaxSize().verticalScroll(rememberScrollState()).padding(16.dp)) {
        val d = data
        when {
            error != null -> Text(error!!, fontSize = 13.sp, color = Color(0xFFDC2626))
            d == null -> LoadingSpinner("Loading game data…")
            else -> {
                StatCard("Pass rate by category") {
                    val series = d.gameSeries.take(6).map { s ->
                        ChartSeries(prettySkillName(s.name), s.data.map { it.toFloat() })
                    }.filter { s -> s.values.any { it > 0f } }
                    if (series.isEmpty()) {
                        Text("No games scored yet. Accuracy by category lights up here when game data arrives.",
                            fontSize = 12.sp, color = Brand.muted)
                    } else {
                        MultiLineChart(series, d.labels, yMax = 100f)
                    }
                }
                Spacer(Modifier.height(14.dp))
                // Each mode is a qualitatively different measurement — never
                // aggregate matching with expressive naming (PRD §5.1).
                StatCard("Pass rate by game mode") {
                    val series = d.modeSeries.map { s ->
                        ChartSeries(prettySkillName(s.name), s.data.map { it.toFloat() })
                    }.filter { s -> s.values.any { it > 0f } }
                    if (series.isEmpty()) {
                        Text("No matching/slideshow/auditory/expressive sessions yet. Each mode gets its own line here.",
                            fontSize = 12.sp, color = Brand.muted)
                    } else {
                        MultiLineChart(series, d.labels, yMax = 100f, chartHeight = 200.dp)
                    }
                }
            }
        }
    }
}

// ── Mastery bars + recent sessions ──────────────────────────────────────────

@Composable
private fun MasterySessionsPage() {
    val c = LocalAppContainer.current
    var data by remember { mutableStateOf<AnalyticsResponse?>(null) }
    var error by remember { mutableStateOf<String?>(null) }

    LaunchedEffect(Unit) {
        try { data = c.api.analytics(c.auth.childSlug) }
        catch (e: Exception) { error = "Could not load: ${e.message}" }
    }

    Column(Modifier.fillMaxSize().verticalScroll(rememberScrollState()).padding(16.dp)) {
        val d = data
        when {
            error != null -> Text(error!!, fontSize = 13.sp, color = Color(0xFFDC2626))
            d == null -> LoadingSpinner("Loading…")
            else -> {
                StatCard("Mastery · last 30 days") {
                    if (d.mastery.isEmpty()) {
                        Text("No game data yet. Start a game and this fills in.",
                            fontSize = 12.sp, color = Brand.muted)
                    }
                    d.mastery.forEach { r ->
                        Row {
                            Text(prettySkillName(r.name), fontSize = 14.sp,
                                fontWeight = FontWeight.SemiBold, color = Brand.ink,
                                modifier = Modifier.weight(1f))
                            Text("${r.pct}%", fontSize = 12.sp, color = Brand.muted)
                        }
                        Spacer(Modifier.height(4.dp))
                        ShareBar(
                            fraction = r.pct / 100f,
                            color = if (r.pct >= 80) hexColor("#10b981") else Brand.pink,
                            height = 10.dp,
                        )
                        Spacer(Modifier.height(10.dp))
                    }
                }
                Spacer(Modifier.height(14.dp))
                StatCard("Recent sessions") {
                    if (d.recentSessions.isEmpty()) {
                        Text("No sessions yet.", fontSize = 12.sp, color = Brand.muted)
                    }
                    d.recentSessions.take(20).forEach { s ->
                        Row(Modifier.padding(vertical = 6.dp)) {
                            Column(Modifier.weight(1f)) {
                                Text(s.mode ?: "Game", fontSize = 14.sp,
                                    fontWeight = FontWeight.SemiBold, color = Brand.ink)
                                Text(listOfNotNull(s.category, s.date).joinToString(" · "),
                                    fontSize = 12.sp, color = Brand.muted)
                            }
                            Column(horizontalAlignment = Alignment.End) {
                                Text(s.result ?: "—", fontSize = 14.sp,
                                    fontWeight = FontWeight.Bold, color = Brand.ink)
                                Text(s.length ?: "", fontSize = 11.sp, color = Brand.muted)
                            }
                        }
                        HorizontalDivider(color = hexColor("#f1e3ec"))
                    }
                }
            }
        }
    }
}
